mongodb_store_msgs/srv/MongoQueryMsg.srv

```diff
@@ -17,6 +17,8 @@ StringPairList message_query
 StringPairList meta_query
 # key/values to be turned into a sort query
 StringPairList sort_query
+# key/values to be turned into a projection query
+StringPairList projection_query
```

Comment:
Yes I think the backward compatibility will be affected. I will try to change this into a new service definition as you have suggested.